pkg/utils/resources/resources.go

```diff
@@ -93,21 +95,84 @@ func Subtract(lhs, rhs v1.ResourceList) v1.ResourceList {
 	return result
 }
 
-// Ceiling calculates the max between the sum of container resources and max of initContainers
-func Ceiling(pod *v1.Pod) v1.ResourceRequirements {
-	var resources v1.ResourceRequirements
+// inspired from https://github.com/kubernetes/kubernetes/blob/e2afa175e4077d767745246662170acd86affeaf/pkg/api/v1/resource/helpers.go#L96
+// https://kubernetes.io/blog/2023/08/25/native-sidecar-containers/
+// Ceiling calculates the max between the sum of container resources and max of initContainers along with sidecar feature consideration
+func podRequests(pod *v1.Pod) v1.ResourceList {
+	requests := v1.ResourceList{}
+	restartableInitContainerReqs := v1.ResourceList{}
+	maxInitContainerReqs := v1.ResourceList{}
+
 	for _, container := range pod.Spec.Containers {
-		resources.Requests = MergeInto(resources.Requests, MergeResourceLimitsIntoRequests(container))
-		resources.Limits = MergeInto(resources.Limits, container.Resources.Limits)
+
+		requests = MergeInto(requests, MergeResourceLimitsIntoRequests(container))
 	}
+
 	for _, container := range pod.Spec.InitContainers {
-		resources.Requests = MaxResources(resources.Requests, MergeResourceLimitsIntoRequests(container))
-		resources.Limits = MaxResources(resources.Limits, container.Resources.Limits)
+
+		if lo.FromPtr(container.RestartPolicy) == v1.ContainerRestartPolicyAlways {
+			requests = MergeInto(requests, MergeResourceLimitsIntoRequests(container))
```

Comment:
Do we need to call `MergeResourceLimitsIntoRequests` twice or does it makes sense to call this at the top of the range loop since this is needed in both cases?